config/routes.rb

```diff
@@ -0,0 +1,23 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  #RESTful routes
+  root to:'homepages#index'
+
+  resources :users
+  resources :votes
+  resources :works
+
+  resources :works do
+    resources :votes
+  end
```

Comment:
You have works twice here and you've defined all the routes for votes.

I suggest:

```suggestion
  resources :works do
    resources :votes, only: [:create]
  end
```